llama_cpp/server/types.py

```diff
@@ -228,6 +237,10 @@ class CreateChatCompletionRequest(BaseModel):
     min_p: float = min_p_field
     stop: Optional[Union[str, List[str]]] = stop_field
     stream: bool = stream_field
+    stream_options: Optional[llama_cpp.StreamOptions] = Field(
+        default=None,
+        description="Options for streaming response. Only set this when you set stream: true.",
+    )
```

Comment:
Presumably this is where you would make use of the description you have in `include_usage_field` somehow?
I wonder if [FastAPI dependency injection](https://fastapi.tiangolo.com/tutorial/dependencies/) could help with this, or if there is an example pattern in the server code for providing property help descriptions that is better followed instead. I've not used FastAPI zealously enough to know yet.
@tpfau

Reply:
I'm not sure how to do this neither. The problem I have is that llama_types defines `TypedDicts` which don't have descriptions like `BaseModels`. So if anyone got a good suggestion I'm happy to update this.